### PR TITLE
doc: add process for adding GitHub Apps

### PR DESCRIPTION
**Pull request by mhdawson**
Signed-off-by: Michael Dawson <michael_dawson@ca.ibm.com>

### CODE REVIEW

**Comment by Trott** [comment]
If we're going to go down this route, I'd prefer to just add it as a bullet point under "GitHub repositories" since it seems to use the same process as those items.

I'm not sure we do want to go down this route, though? GitHub Apps reviews usually take quite a bit longer than 72 hours, and I think that's appropriate. Maybe it's not and I'm part of the problem or something, but I think it's OK to spend time really getting to understand what an App will do before enabling it. This isn't the same as creating a repository or something like that.

**Comment by Trott** [comment]
Maybe all that needs to happen is removing the auto-approval part? That basically would end up being documentation of our current process, I think. We can then subsequently talk about if adding auto-approval makes sense?

**Comment by bnb** [comment]
>A new request is required each time an application is enabled for a new repository even if it has already been installed.

I'd like to ask: for WG, Team, and Initiative repos is full TSC and CommComm approval really necessary?

If the maintainers of nodejs/nodejs.org want to enable dependabot, codecov, or any other installed app and agree as a WG / initiative / teams to do so, I don't really see a need for that to go through an approval in nodejs/admin if the app has already been added to the org for a different repo.

Seems to add a boatload of overhead both for the individual repo maintainers and the TSC / CommComm for no direct or obvious benefit?

**Comment by MylesBorins** [comment]
Adding an app to the org can give it all sorts of different privledges,
they are not necessarily scoped to the repo.

Perhaps we don't need fill sign off, but at the very least we need to
delegate the vetting of the application and it can't be approved by the
group requesting it.

I think, but could be wrong, that the TSC will likely want to maintain or
at the very least delegate the authority to verify and do risk analysis of
each addition

**Comment by MylesBorins** [comment]
Also, it should be possible to do a broader review of an application once
and have a blanket approval for future use.

I think of it more as security / privacy review fwiw

**Comment by bnb** [comment]
>Perhaps we don't need fill sign off, but at the very least we need to delegate the vetting of the application and it can't be approved by the group requesting it.

>Also, it should be possible to do a broader review of an application once and have a blanket approval for future use.

This is definitely in-line with what I was thinking.

If our concern is org permission and impact, I don't think gating folks who want to use apps that are already in the org is going to be meaningfully helpful *if* we turn on the "only select repositories" feature for each app, since the org permissions would have already been granted and nothing would be meaningfully changing beyond the introduction of the app into a repo that wants it.

**Comment by mhdawson** [comment]
I'm, ok with removing the auto approval

**Comment by mhdawson** [comment]
@Trott pushed an update to remove auto approval.

**Comment by mhdawson** [comment]
Tweaked a bit more based on discussion so far.

**Comment by mhdawson** [comment]
@MylesBorins 
> Also, it should be possible to do a broader review of an application once
and have a blanket approval for future use. 

I think that can/should be covered in a follow on PR as an extension as it likely requires a different level of review for blanket future use.

Are you ok with the current content in terms of the due diligence required for approval for specific repositories?

File: GITHUB_ORG_MANAGEMENT_POLICY.md [hunk 1]
```diff
@@ -88,6 +88,28 @@ Blocking an individual who is not currently a member of the GitHub organization
...
+issue in the [Node.js admin repository][nodejs/admin] with details of:
+
+* the application to be installed
+* the permissions that it requires
+* the repositories for which it will be enabled. 
+
+A new request is required each time an application is enabled
+for a new repository even if it has already been installed.
+
+The request must be approved by at least two TSC or CommComm members and
```
Comment (Trott) [inline GITHUB_ORG_MANAGEMENT_POLICY.md:106]:
```suggestion
The request must be approved by at least two TSC and two CommComm members and
```

**Comment by mhdawson** [comment]
Believe comments have been addressed and there are enough approvers landing.

**Pull request merged**